samples/template/README.rst

```diff
@@ -0,0 +1,167 @@
+This solution is required only for the Device Firmware Update (DFU).
+
+
+Wiring*
+*******
+
+TODO: semtech wiring
+
+User interface*
+***************
+
+Button 1:
+   Send Hello - This action will queue a message to the cloud.
+   If Sidewalk is not ready, it will simply show an error without sending the message.
+   The queue will be processed eventually, and all the queued messages will be sent.
+   On BLE application performs Connection Requests before sending automatically if necessary.
+
+Button 2:
+   Toggles link mask. BLE/FSK/LoRa
+
+Button 4:
+   Factory reset - The application informs the Sidewalk stack about the factory reset event.
+   The Sidewalk library clears its configuration from the non-volatile storage.
+   After a successful reset, the device needs to be registered with the cloud services again.
+
+LED 1:
+   On when not connected. Sidewalk started but not ready to send messages.
+
+LED 2:
+   On when connected. Sidewalk is ready to send a message.
+
+
+
+Configuration*
+**************
+
+When running the sample, you can use different configurations files depending on the supported features.
+
+.. _template_subghz_build_type:
+
+Build types
+===========
+
+Configuration files are provided for different build types and are located in the application root directory.
+
+The :file:`prj.conf` file represents a ``debug`` build type.
+Other build types are covered by dedicated files with the respective build type added to the file name.
+For example, the ``release`` build type file is named :file:`prj_release.conf`.
+The same naming convention is followed if a board has other configuration files that are, for example, associated with the partition layout or child image configuration.
+
+Before you start testing the application, you can select one of the build types supported by the sample.
+Depending on the selected board, a sample supports the following build types:
+
+* ``debug`` -- Debug version of the application - can be used to enable additional features, such as logs or command-line shell, to verify the application behavior.
+* ``release`` -- Release version of the application - can be used to enable only the necessary application functionalities to optimize its performance.
+* ``no_dfu`` -- No dfu version of the application - can be used in simple scenarios where dfu is not required. Can also be a starting point to integrate sidewalk sample with custom bootloader.
+
+You can build the ``release`` firmware for ``build_target`` by running the following command in the project directory:
+
+.. parsed-literal::
+   :class: highlight
+
+   $ west build -b *build_target* -- -DCONF_FILE=prj_release.conf
+
+For example:
+
+.. code-block:: console
+
+   $ west build -b nrf52840dk_nrf52840 -- -DCONF_FILE=prj_release.conf
+
+.. note::
+    Selecting a build type is optional.
+    However, if the build type is not selected, the ``debug`` build type is used by default.
```

Comment:
yes, but as I'm aware we still use size optimization in this configuration, so maybe we should use the debug optimization in this scenario ? 

I see this difference when I start debugging session, when I build with VSCode and select debug optimization I see all threads in the debug, but when I build from console, the threads are not separated in debugger

Reply:
blocked by https://github.com/nrfconnect/sdk-nrf/pull/13621